### PR TITLE
Persistence Component: Introduce Postgres Sink

### PR DESCRIPTION
**Pull request by kumuwu**
#### What type of PR is this?

- Improvement
- Bug Fix

#### What does this PR do / why is it needed ?

Introduce Postgres sink support in library.
Ingest modes supported:

- Unitemporal Snapshot
- Append Only

Also fixed bug in length not being set in data types.
#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/SqlUtils.java [hunk 1]
```diff
@@ -55,6 +55,7 @@ public static void logSql(Logger logger, SqlLogging sqlLogging, String sqlBefore
                 break;
             case UNMASKED:
                 logger.info(sqlAfterReplacingPlaceholders);
+                System.out.println(sqlAfterReplacingPlaceholders);
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/SqlUtils.java:58]: Let's remove the Sysouts.

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/executor/RelationalExecutor.java [hunk 2]
```diff
@@ -135,7 +135,7 @@ public Dataset constructDatasetFromDatabase(Dataset dataset)
     @Override
     public void setSqlLogging(SqlLogging sqlLogging)
     {
-        this.sqlLogging = sqlLogging;
+        this.sqlLogging = SqlLogging.UNMASKED;
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/executor/RelationalExecutor.java:138]: Is this for testing ?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/pom.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+            <version>3.10</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>testcontainers</artifactId>
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/pom.xml:73]: Can we keep the scope as test if it's not set in the parent ?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/pom.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>postgresql</artifactId>
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/pom.xml:77]: Can we keep the scope as test if it's not set in the parent ?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/main/java/org/finos/legend/engine/persistence/components/relational/postgres/PostgresSink.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+            return new RelationalExecutor(this, JdbcHelper.of(jdbcConnection.connection()));
+        }
+        else
+        {
+            throw new UnsupportedOperationException("Only JdbcConnection is supported for Postgres Sink");
+        }
+    }
+
+    @Override
+    public Optional<Optimizer> optimizerForCaseConversion(CaseConversion caseConversion)
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/main/java/org/finos/legend/engine/persistence/components/relational/postgres/PostgresSink.java:152]: Possible to move this to AnsiSink for all the sinks

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/SqlUtils.java (hunk 1 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/SqlUtils.java:58]: Sure, removed - was used in testing and accidentally committed

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/executor/RelationalExecutor.java (hunk 2 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/executor/RelationalExecutor.java]: Sure, removed - was used in testing and accidentally committed

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/pom.xml (hunk 3 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/pom.xml:83]: Sure, changed

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/main/java/org/finos/legend/engine/persistence/components/relational/postgres/PostgresSink.java (hunk 5 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/main/java/org/finos/legend/engine/persistence/components/relational/postgres/PostgresSink.java]: Sure, changed! Removed the overriding method from Postgres, H2, Bigquery and Memsql; retained the one in Snowflake as it is using a different optimizer

**Review by epsstan — APPROVED** [review]

**Pull request merged**